pkg/restore/db.go

```diff
@@ -210,12 +210,13 @@ func FilterDDLJobs(allDDLJobs []*model.Job, tables []*utils.Table) (ddlJobs []*m
 		tableNames[table.Info.Name.String()] = true
 		for _, job := range allDDLJobs {
 			if job.BinlogInfo.TableInfo != nil {
-				if tableIDs[job.TableID] || tableNames[job.BinlogInfo.TableInfo.Name.String()] {
+				name := job.SchemaName + job.BinlogInfo.TableInfo.Name.String()
```

Comment:
🤔 

Can we use a `struct { schemaName string; tableName string }` as the key? The current scheme will still break for `"abc" + "de"` vs `"ab" + "cde"`.